content/en/docs/concepts/workloads/pods/init-containers.md

```diff
@@ -289,6 +289,31 @@ The Pod which is already running correctly would be killed by `activeDeadlineSec
 The name of each app and init container in a Pod must be unique; a
 validation error is thrown for any container sharing a name with another.
 
+### Sidecar containers and restartPolicy
+
+{{< feature-state for_k8s_version="v1.28" state="alpha" >}}
+
+Thanks to a newly available feature gate named `SidecarContainers`, each
+init container can have its own `restartPolicy`, independent of the Pod and the
+other init containers, as well as any regular [probe](/docs/concepts/workloads/pods/pod-lifecycle/#types-of-probe).
+
+If an init container is created with a `restartPolicy` set to `Always` and either
+a `startupProbe` or a `readinessProbe`, it will start and remain running during the
+entire life of the Pod, which is useful for running supporting services isolated
+from the main application container.
+
+Since these containers are defined as init containers, they benefit from the same
+ordering and sequential guarantees as other init containers, allowing them to
+be mixed with other init containers into complex Pod initialization flows.
+
+The only difference is that they are not required to complete before the next
+init container starts, which is impossible, so next init container will start
+after the `startupProbe` or `readinessProbe` of the container has succeeded.
+
+This feature can be used to implement the sidecar container pattern in a more
+robust way, as the sidecar container will be restarted if it fails directly
+by the kubelet, instead of relying on the application container to restart it.
+
```

Comment:
Might mention how it makes it possible to use sidecars with Jobs as well. That's a use case that's not easy without this feature.